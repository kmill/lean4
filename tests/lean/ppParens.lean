set_option pp.parens true

#check 1 + 2 * 3 + 4

#check (Nat.add_assoc)

#check [1,2,3] ++ [3,4,5]

#check ∀ (p : Nat → Nat → Prop), p (1 + 2) 3

#check if True then 1 else 2
