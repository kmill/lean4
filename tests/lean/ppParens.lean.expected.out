(((1) + ((2) * (3))) + (4)) : (Nat)
(Nat.add_assoc) : (∀ (n m k : (Nat)), ((((n) + (m)) + (k)) = ((n) + ((m) + (k)))))
(([(1), (2), (3)]) ++ ([(3), (4), (5)])) : ((List) (Nat))
(∀ (p : ((Nat) → ((Nat) → (Prop)))), ((p) ((1) + (2)) (3))) : (Prop)
(if (True) then (1) else (2)) : (Nat)
